pinot-core/src/main/java/org/apache/pinot/core/data/manager/BaseTableDataManager.java

```diff
@@ -102,6 +103,10 @@ public void init(TableDataManagerConfig tableDataManagerConfig, String instanceI
     if (!_indexDir.exists()) {
       Preconditions.checkState(_indexDir.mkdirs());
     }
+    _resourceTmpDir = new File(_indexDir, "tmp");
```

Comment:
So with this approach we
1) move the tmp files into their own tmp folder
2) create the tmp folder nested under the table's data dir

@klsince, how do you feel about this approach

Reply:
cc @mayankshriv wdyt?

this lgtm. perhaps use "_tmp" here so that no need to touch the HLR.../LLR.. subclasses

Reply:
sgtm. `tmp` is fine.

perhaps leave a comment that this cleanup logic handles tmp resource from TableDataManager specifically, and tmp resources left by others may be cleaned as well if they happen to be put at tmp folder under table data dir.

Reply:
added a comment